### PR TITLE
API: /oidc/register api definition

### PR DESCRIPTION
**Pull request by sebbalex**
fix: /oidc/register api definition
feat: /oidc/register/client_id api definition

### CODE REVIEW

File: src/infra/dev/api/oi.tpl.json [hunk 1]
```diff
@@ -289,29 +289,634 @@
...
+        "operationId": "Post_oidc_register",
         "x-amazon-apigateway-integration" : {
+          "passthroughBehavior" : "when_no_match",
+          "contentHandling" : "CONVERT_TO_TEXT",
+          "type" : "aws_proxy",
           "httpMethod" : "POST",
           "uri" : "arn:aws:apigateway:${aws_region}:lambda:path/2015-03-31/functions/${client_registration_lambda_arn}/invocations",
-          "responses" : {},
+          "requestParameters": {
+            "integration.request.header.accept": "'*/*'"
```
Comment (uolter) [inline src/infra/dev/api/oi.tpl.json:371]: Should the cors header reach the backend (lambda ) too?

File: src/infra/dev/api/oi.tpl.json [hunk 2]
```diff
@@ -289,29 +289,634 @@
...
+                  },
+                  "environment": {
+                    "type": "string",
+                    "example": "ALL"
+                  },
+                  "master": {
+                    "type": "boolean",
+                    "example": false
+                  },
+                  "spid_requested_attributes": {
```
Comment (GiuMontesano) [inline src/infra/dev/api/oi.tpl.json:484]:
```suggestion
                  "saml_requested_attributes": {
```

File: src/infra/dev/api/oi.tpl.json [hunk 3]
```diff
@@ -289,29 +289,634 @@
...
+                    "type": "string",
+                    "example": "web"
+                  },
+                  "contacts": {
+                    "type": "array",
+                    "items": {
+                      "type": "string"
+                    },
+                    "example": ["user1@example.org", "user2@example.org"]
+                  },
```
Comment (GiuMontesano) [inline src/infra/dev/api/oi.tpl.json:447]:
```suggestion

```

File: src/infra/dev/api/oi.tpl.json [hunk 4]
```diff
@@ -289,29 +289,634 @@
...
+                    "example": "Registered via OpenID Connect Dynamic Registration"
+                  },
+                  "environment": {
+                    "type": "string",
+                    "example": "ALL"
+                  },
+                  "master": {
+                    "type": "boolean",
+                    "example": false
+                  },
```
Comment (GiuMontesano) [inline src/infra/dev/api/oi.tpl.json:483]:
```suggestion

```

File: src/infra/dev/api/oi.tpl.json [hunk 5]
```diff
@@ -289,29 +289,634 @@
...
+                    "description": "Custom parameter which return all SPID attributes selected, https://docs.italia.it/italia/spid/spid-regole-tecniche/it/stabile/attributi.html",
+                    "example": ["spidCode", "name", "familyName", "fiscalNumber"]
+                  }
+                }
+              }
+            }
+          },
+          "description": "Request body",
+          "required": true
+        },
```
Comment (GiuMontesano) [inline src/infra/dev/api/oi.tpl.json:498]: We also need, from the documentation:
- URI logo_uri;
- List<String> default_acr_values;

**Review by GiuMontesano — CHANGES_REQUESTED** [review]

File: src/infra/dev/api/oi.tpl.json [hunk 6]
```diff
@@ -289,29 +289,634 @@
...
+                    "registration_access_token": {
+                      "type": "string",
+                      "example": "db4b7554-e359-43ed-bbf6-1a50909ceeef",
+                      "description": "An access_token to be used at {registration_client_uri} to retrieve the registered information"
+                    },
+                    "registration_client_uri": {
+                      "type": "string",
+                      "example": "https://server.example.org/connect/register?client_id=s6BhdRkqt3",
+                      "description": "The URI to retrieved the registered values"
+                    },
```
Comment (giuseppe-gangemi) [inline src/infra/dev/api/oi.tpl.json:531]: given that this is implemented using the path parameter, this should be deleted, right?

File: src/infra/dev/api/oi.tpl.json (hunk 1 above)
Comment (sebbalex) [reply src/infra/dev/api/oi.tpl.json:371]: I guess they could be handled by the apigw itself

File: src/infra/dev/api/oi.tpl.json (hunk 6 above)
Comment (sebbalex) [reply src/infra/dev/api/oi.tpl.json:531]: my bad, going to fix it

File: src/infra/dev/api/oi.tpl.json (hunk 3 above)
Comment (sebbalex) [reply src/infra/dev/api/oi.tpl.json:447]: did not get it, is it just a tabulation?

File: src/infra/dev/api/oi.tpl.json (hunk 4 above)
Comment (sebbalex) [reply src/infra/dev/api/oi.tpl.json:483]: same here

**Review by uolter — APPROVED** [review]

File: src/infra/dev/api/oi.tpl.json (hunk 3 above)
Comment (GiuMontesano) [reply src/infra/dev/api/oi.tpl.json:447]: No. These are fields not present Java-side

File: src/infra/dev/api/oi.tpl.json (hunk 4 above)
Comment (GiuMontesano) [reply src/infra/dev/api/oi.tpl.json:483]: Same

File: src/infra/dev/api/oi.tpl.json (hunk 3 above)
Comment (sebbalex) [reply src/infra/dev/api/oi.tpl.json:447]: ops, sorry

File: openapi/oi.json [hunk 7]
```diff
@@ -646,74 +675,40 @@
...
                     "items": {
                       "type": "string"
                     },
-                    "description": "Custom parameter which return all SPID attributes selected, https://docs.italia.it/italia/spid/spid-regole-tecniche/it/stabile/attributi.html",
-                    "example": ["spidCode", "name", "familyName", "fiscalNumber"]
+                    "description": "list of the Authentication Context Class References that this RP supports",
+                    "example": [
+                      "https://www.spid.gov.it/SpidL1",
+                      "https://www.spid.gov.it/SpidL2",
+                      "https://www.spid.gov.it/SpidL3"
```
Comment (GiuMontesano) [inline openapi/oi.json:710]:
```suggestion
                      "L1",
                      "L2",
                      "L3"
```
For now we expose these values via an Enum. Let us reason together whether we prefer the whole string over just the levels.

**Review by GiuMontesano — CHANGES_REQUESTED** [review]

**Review by GiuMontesano — APPROVED** [review]
LGTM

**Pull request merged**